### PR TITLE
Defer support

### PR DESCRIPTION
**Pull request by trevor-scheer**
Implements support for experimental `@defer` usage.

This uses the raw response object to write to and end the stream. There isn't an obvious (to me) API provided by Koa for multi-part responses like this.

In order to try this out, users must install the `graphql@17` canary build into their AS4 project like so:
`npm i --legacy-peer-deps graphql@17.0.0-alpha.1.canary.pr.3361.04ab27334641e170ce0e05bc927b972991953882`

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -97,9 +97,14 @@ export function koaMiddleware<TContext extends BaseContext>(
       if (body.kind === 'complete') {
         ctx.body = body.string;
         return;
+      } else if (body.kind === 'chunked') {
+        for await (const chunk of body.asyncIterator) {
+          ctx.response.res.write(chunk);
```
Comment (matthew-gordon) [inline src/index.ts:102]: ahh nice 🙏 💪

**Comment by laverdet** [comment]
Two things:
- You should add `ctx.respond = false` which tells Koa not to respond since you're doing the low-level work. Otherwise once this function returns Koa will try to do `res.write("OK")`. https://github.com/koajs/koa/blob/master/docs/api/context.md#ctxrespond
- Also, a conditional `flush` invocation should be added which will ensure if a compression middleware [gzip, brotli] is used then the data stream won't be buffered. This is done in the core Apollo express middleware: https://github.com/apollographql/apollo-server/blob/a69580565dadad69de701da84092e89d0fddfa00/packages/server/src/express4/index.ts#L96-L105

Otherwise I can confirm that this works great in initial testing. I was shocked to see that this feature wasn't already supported.

**Comment by trevor-scheer** [comment]
@laverdet if you can confirm that the approach including your suggestions is kosher / correct for koa I'd be happy for you to make a new PR off of my branch. I'll be offline for the week.

The reason this PR is stalled is because there might be a more correct approach with using actual streams that I didn't figure out or have the time to continue investigating.

But I suppose that's maybe an implementation that we could always cut over to so long as tests are passing.

In any case feel free to push this work forward if you feel inclined to!

**Comment by trevor-scheer** [comment]
Superseded by #85

**Pull request closed without merging**